### PR TITLE
assign-labels.sh: lint

### PR DESCRIPTION
**Pull request by HaxxonHax**
Changelog:
  - Fix expanding an array without an index only gives the first element.
  - Double quoted variables to prevent globbing and word splitting.

### CODE REVIEW

File: ci/assign-labels.sh [hunk 1]
```diff
@@ -49,7 +49,7 @@ label_issue() {
   fi
 
   if [[ ${DRY_RUN} != true ]]; then
-    send_gh_api_data_request repos/${NEOVIM_REPO}/issues/${issue_id} \
+    send_gh_api_data_request repos/"${NEOVIM_REPO}"/issues/"${issue_id}" \
```
Comment (justinmk) [inline ci/assign-labels.sh]: is shellcheck happy with this instead:

```suggestion
    send_gh_api_data_request "repos/${NEOVIM_REPO}/issues/${issue_id}" \
```

File: ci/assign-labels.sh [hunk 2]
```diff
@@ -33,8 +33,8 @@ label_issue() {
     # Remove existing mutually-exclusive labels, if any.
     if is_label_exclusive "$new_label"; then
       local i
```
Comment (justinmk) [inline ci/assign-labels.sh]: is this needed now?

File: ci/assign-labels.sh [hunk 3]
```diff
@@ -2,10 +2,10 @@
...
 
 BUILD_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
-source ${BUILD_DIR}/ci/common/common.sh
-source ${BUILD_DIR}/ci/common/dependencies.sh
-source ${BUILD_DIR}/ci/common/github-api.sh
-source ${BUILD_DIR}/ci/common/neovim.sh
+source "${BUILD_DIR}"/ci/common/common.sh
+source "${BUILD_DIR}"/ci/common/dependencies.sh
+source "${BUILD_DIR}"/ci/common/github-api.sh
+source "${BUILD_DIR}"/ci/common/neovim.sh
```
Comment (justinmk) [inline ci/assign-labels.sh]:
```suggestion
source "${BUILD_DIR}/ci/common/neovim.sh"
```

File: ci/assign-labels.sh (hunk 1 above)
Comment (HaxxonHax) [reply ci/assign-labels.sh]: Yes,, shellcheck is happy with this.  I have seen some shells not like this expansion for file paths.  As long as bash will always be used, this should be fine.

File: ci/assign-labels.sh (hunk 2 above)
Comment (HaxxonHax) [reply ci/assign-labels.sh]: It does not appear to be referenced within the context.  Shellcheck did not pick this up.

File: ci/assign-labels.sh (hunk 3 above)
Comment (HaxxonHax) [reply ci/assign-labels.sh]: Canonical changes added to new commit.

File: ci/assign-labels.sh (hunk 2 above)
Comment (HaxxonHax) [reply ci/assign-labels.sh]: Removed in new commit.

**Pull request merged**